crates/kornia-tensor-ops/src/ops.rs

```diff
@@ -254,10 +446,246 @@ where
     Ok(T::one() - similarity)
 }
 
+/// Trait providing tensor operations for CPU-based tensors.
+///
+/// This trait defines a collection of mathematical operations that can be performed on tensors.
+/// It serves as an interface for implementing common tensor operations such as element-wise
+/// operations (addition, multiplication, etc.), reductions (sum, mean), and transformations.
+///
+/// The operations exposed by this trait are implemented as static methods that take a reference
+/// to the tensor as their first argument, following a functional programming style.
+///
+/// # Type Parameters
+///
+/// * `T` - The data type of the tensor elements.
+/// * `N` - The number of dimensions (rank) of the tensor as a const generic.
+///
+/// # Implementation
+///
+/// This trait is implemented for `Tensor<T, N, CpuAllocator>` where `T` implements `Float`,
+/// meaning these operations are available for floating-point tensors on CPU.
+///
+/// # Examples
+///
+/// ```
+/// use kornia_tensor::{Tensor, CpuAllocator};
+/// use kornia_tensor_ops::ops::TensorOps;
+///
+/// // Create a tensor
+/// let data = vec![1.0, 2.0, 3.0, 4.0];
+/// let t = Tensor::<f32, 2, CpuAllocator>::from_shape_vec([2, 2], data, CpuAllocator).unwrap();
+///
+/// // Use operations through the trait
+/// let scaled = Tensor::mul_scalar(&t, 2.0);
```

Comment:
```suggestion
/// let scaled = t.mul_scalar(2.0);
```
this should be the ideal usage. Same for the others